livre/core/Render/RenderBrick.cpp

```diff
@@ -105,6 +105,12 @@ void RenderBrick::getScreenCoordinates( const Frustum& frustum,
                     continue;
                 in.normalize();
 
+                const double perspCorr = 1.0 / in.w();
+
+                in[0] *= perspCorr;
+                in[1] *= perspCorr;
+                in[2] *= perspCorr;
+
```

Comment:
Ok. But cannot be const. We are modifying it:
in[0] *= perspCorr and in[0] = in[0] * 0.5 + 0.5;